rfcs/0001-shabka-usage-design.md

```diff
@@ -0,0 +1,186 @@
+---
+feature: shabka-usage-design
+start-date: 2019-10-14
+author: Marc 'risson' Schmitt
+co-authors:
+related-issues: https://github.com/kalbasit/shabka/projects/10 https://github.com/kalbasit/shabka/issues/263 https://github.com/kalbasit/shabka/issues/246 https://github.com/kalbasit/shabka/issues/55 https://github.com/kalbasit/shabka/issues/281
+---
+
+# Summary
+[summary]: #summary
+
+The goal is to design a system where `shabka` is just a tool like
+`home-manager`, and relies on configuration files provided by the user.
+The tools and ways to configure at the disposition of the user would be those:
+
+* the command `shabka`, installed like `home-manager` which basically does the
+  stuff the user wants to do on a regular basis that we currently have in
+  `./scripts` ;
+* a `dotshabka` like directory for the user's personal preferences settings and
+  default values for `shabka` options, or overriding stuff we define in
+  `shabka`. We will refer to this as `dotshabka-user` ;
+* a `dotshabka` like directory for the user's hosts. We will refer to this as
+  `.shabka`.
```

Comment:
When I initially read this, I left a comment about using one directory. But then I read the section below (`## .shabka for hosts configuration`) and it turns out you did intend to use one directory. Do we really need then two parameters `.shaba` and `dotshabka-user`? Did I miss something?

Reply:
Here's what we would have following this design for a single user installation:
* shabka command
* `.config/shabka` containing:
  - hosts
  - user global personal configuration (what's in `dotshabka/modules`)
  - user per-host personal configuration
But for a multi-user installation, we would have several of those last two entries. `user global personal configuration` would be once per user. `user per-host personal configuration` would be once per user per host.
Hence, I think we could put the hosts in one repository, and the user configurations in another repository.